components/mpas-ocean/src/shared/mpas_ocn_vmix_cvmix.F

```diff
@@ -271,41 +279,122 @@ subroutine ocn_vmix_coefs_cvmix_build(meshPool, statePool, forcingPool, err, tim
+                 sigma = -zt_cntr(kIndexOBL,iCell) / OBLDepths(kIndexOBL)
+
+                 interfaceForcings(kIndexOBL) = surfaceBuoyancyForcing(iCell)
+
+                 ! initialize the surfaceAverageIndex for cases when the if statement below is not true
+                 surfaceAverageIndex(kIndexOBL,iCell) = minLevelCell(iCell)
+                 ! move progressively downward to find the bottom most layer within the surface layer
+                 sfc_layer_depth = OBLDepths(kIndexOBL) * config_cvmix_kpp_surface_layer_extent
+                 do kav=topIndex,kIndexOBL
+                    if(zw_iface(kav+1,iCell) < -sfc_layer_depth) then
+                       surfaceAverageIndex(kIndexOBL,iCell) = kav
+                       exit
+                    end if
+                 enddo
+                 topIndex = max(1, surfaceAverageIndex(kIndexOBL,iCell) - 1)
+              end do
+              stable = 0.5_RKIND + SIGN(0.5_RKIND,surfaceBuoyancyForcing(iCell))
+              sigma = stable + (1.0_RKIND - stable)*config_cvmix_kpp_surface_layer_extent
+              ! compute the turbulent scales in order to compute the bulk Richardson number
+              call cvmix_kpp_compute_turbulent_scales( &
+                   sigma_coord = sigma, &
+                   OBL_depth = OBLDepths(minLevelCell(iCell):maxLevelCell(iCell)), &
+                   surf_buoy_force = interfaceForcings(minLevelCell(iCell):maxLevelCell(iCell)), &
+                   surf_fric_vel = surfaceFrictionVelocity(iCell), &
+                   w_s = turbulentScalarVelocityScale(minLevelCell(iCell):maxLevelCell(iCell),iCell))
+
+                 !redo me for T&S
+                 !check all variale declarations
+              do iTracer=1,2
+                 tracerTmp(minLevelCell(iCell)) = activeTracers(iTracer,minLevelCell(iCell), iCell)* &
+                                                         layerThickness(minLevelCell(iCell), iCell)
+                 layerThicknessSum(minLevelCell(iCell)) = layerThickness(minLevelCell(iCell), iCell)
+                 do kIndexOBL = minLevelCell(iCell)+1, maxLevelCell(iCell)
+                    layerThicknessSum(kIndexOBL) = layerThicknessSum(kIndexOBL-1) + layerThickness(kIndexOBL, iCell)
+                    tracerTmp(kIndexOBL) = tracerTmp(kIndexOBL-1) + &
+                                   layerThickness(kIndexOBL, iCell)*activeTracers(iTracer, kIndexOBL, iCell)
+                 end do
+                 do kIndexOBL = minLevelCell(iCell),maxLevelCell(iCell)
+                    tracersSum(iTracer,kIndexOBL,iCell) = tracerTmp(surfaceAverageIndex(kIndexOBL,iCell)) &
+                                    / layerThicknessSum(surfaceAverageIndex(kIndexOBL,iCell))
+                 end do
+              end do
+
+      enddo
+      !with the tracerSum being the sfc average for each, call density eos
+       call ocn_equation_of_state_density(statePool, meshPool, tracersSum, tracersSurfaceValue, nCells, 0, &
```

Comment:
Because the equation of state is being performed on the accelerator when OpenACC is enabled and this routine is not net on the device, you will need:
!$acc enter data copyin(tracersSum, tracersSurfaceValue)
before the acc call and
!$acc update host (sfcParcelDensity)
!$acc exit data copyout(tracersSum, tracersSurfaceValue)
after this call (if sfcParcelDensity is moved from a diagnostic array to a  module private or local subroutine array, then eliminate the update host and add sfcParcelDensity to the copyin/copyout list).